### PR TITLE
Update Slack alerts

### PR DESCRIPTION
**Pull request by SpennyNDaJets**
## Summary
Slack alerts are not standardized and can be hard to decipher. Update the Slack alerts to use different colors to signify level and standardize how they are displayed.

## How did you test this change?
1) Create each of the alert and view in #spencer-error-test channel

## Are there any deployment considerations?
N/A

## Does this work require review from our design team?
Yes

### CODE REVIEW

File: backend/model/model.go [hunk 1]
```diff
@@ -2779,48 +2805,101 @@ func (obj *Alert) sendSlackAlert(ctx context.Context, db *gorm.DB, alertID int,
...
-		blockSet = append(blockSet, slack.NewActionBlock(
-			"",
-			actionBlock...,
-		))
-		blockSet = append(blockSet, slack.NewDividerBlock())
 		msg.Attachments = []slack.Attachment{
 			{
 				Color:  "#961e13",
-				Blocks: slack.Blocks{BlockSet: blockSet},
+				Blocks: slack.Blocks{BlockSet: headerBlockSet},
```
Comment (SpennyNDaJets) [inline backend/model/model.go:2877]: TODO(spenny): test this

File: backend/model/model.go [hunk 2]
```diff
@@ -2647,11 +2649,6 @@ type SendSlackAlertInput struct {
 
 func getUserPropertiesBlock(identifier string, userProperties map[string]string) ([]*slack.TextBlockObject, *slack.Accessory) {
 	messageBlock := []*slack.TextBlockObject{}
-	if identifier != "" {
-		messageBlock = append(messageBlock, slack.NewTextBlockObject(slack.MarkdownType, "*User:*\n"+identifier, false, false))
-	} else {
-		messageBlock = append(messageBlock, slack.NewTextBlockObject(slack.MarkdownType, "*User:*\n_unidentified_", false, false))
-	}
```
Comment (SpennyNDaJets) [inline backend/model/model.go:2654]: Move this to the session block

File: backend/model/model.go [hunk 3]
```diff
@@ -2739,32 +2765,69 @@ func (obj *Alert) sendSlackAlert(ctx context.Context, db *gorm.DB, alertID int,
...
+			sessionUserIdentifier = "_unidentified_ user"
+		}
+
+		sessionLink := fmt.Sprintf("%s/%d/sessions/%s%s", frontendURL, obj.ProjectID, input.SessionSecureID, suffix)
+		sessionText := fmt.Sprintf("#%s (%s)", input.SessionSecureID, sessionUserIdentifier)
+		sessionBlock = slack.NewTextBlockObject(slack.MarkdownType, fmt.Sprintf("*Session* <%s|%s>", sessionLink, sessionText), false, false)
+	}
+
 	switch *obj.Type {
 	case AlertType.ERROR:
```
Comment (SpennyNDaJets) [inline backend/model/model.go:2804]: Screenshot 2023-10-09 at 7 53 37 PM

File: backend/model/model.go [hunk 4]
```diff
@@ -2779,48 +2842,79 @@ func (obj *Alert) sendSlackAlert(ctx context.Context, db *gorm.DB, alertID int,
...
+
+		bodyBlockSet = append(bodyBlockSet, slack.NewSectionBlock(eventBlock, nil, nil))
+		bodyBlockSet = append(bodyBlockSet, slack.NewSectionBlock(errorMessageBlock, nil, nil))
+		bodyBlockSet = append(bodyBlockSet, slack.NewSectionBlock(sessionBlock, nil, nil))
+		bodyBlockSet = append(bodyBlockSet, slack.NewActionBlock("", actionBlocks...))
+		if stackTraceBlock != nil {
+			highlightLogo := *slack.NewImageBlockElement("https://app.highlight.io/logo192.png", "Highlight logo")
+			bodyBlockSet = append(bodyBlockSet, slack.NewContextBlock("", highlightLogo, stackTraceBlock))
 		}
 	case AlertType.NEW_USER:
```
Comment (SpennyNDaJets) [inline backend/model/model.go:2907]: I don't believe this is working - can create a card to investigate and fix separately

File: backend/model/model.go [hunk 5]
```diff
@@ -2779,48 +2842,79 @@ func (obj *Alert) sendSlackAlert(ctx context.Context, db *gorm.DB, alertID int,
...
+		var attributeBlocks []*slack.TextBlockObject
 		userPropertiesBlock, accessory := getUserPropertiesBlock(identifier, input.UserProperties)
-		messageBlock = append(messageBlock, userPropertiesBlock...)
-		blockSet = append(blockSet, slack.NewSectionBlock(textBlock, messageBlock, accessory))
-		blockSet = append(blockSet, slack.NewDividerBlock())
-		msg.Blocks = &slack.Blocks{BlockSet: blockSet}
+		attributeBlocks = append(attributeBlocks, userPropertiesBlock...)
+
+		bodyBlockSet = append(bodyBlockSet, slack.NewSectionBlock(sessionBlock, attributeBlocks, accessory))
 	case AlertType.TRACK_PROPERTIES:
```
Comment (SpennyNDaJets) [inline backend/model/model.go:2918]: Screenshot 2023-10-09 at 7 54 48 PM

File: backend/model/model.go [hunk 6]
```diff
@@ -2831,62 +2925,76 @@ func (obj *Alert) sendSlackAlert(ctx context.Context, db *gorm.DB, alertID int,
...
+		headerBlock := slack.NewTextBlockObject(slack.MarkdownType, "*Track Properties Alert*", false, false)
+		headerBlockSet = append(headerBlockSet, slack.NewSectionBlock(headerBlock, nil, nil))
+
+		// body
+		var attributeBlocks []*slack.TextBlockObject
+		attributeBlocks = append(attributeBlocks, slack.NewTextBlockObject(slack.MarkdownType, fmt.Sprintf("*Matched Track Properties*\n%+v", matchedFormattedFields), false, false))
+		attributeBlocks = append(attributeBlocks, slack.NewTextBlockObject(slack.MarkdownType, fmt.Sprintf("*Related Track Properties*\n%+v", relatedFormattedFields), false, false))
+
+		bodyBlockSet = append(bodyBlockSet, slack.NewSectionBlock(sessionBlock, attributeBlocks, nil))
 	case AlertType.USER_PROPERTIES:
```
Comment (SpennyNDaJets) [inline backend/model/model.go:2939]: Screenshot 2023-10-09 at 7 55 13 PM

File: backend/model/model.go [hunk 7]
```diff
@@ -2831,62 +2925,76 @@ func (obj *Alert) sendSlackAlert(ctx context.Context, db *gorm.DB, alertID int,
...
+		// header
+		headerBlock := slack.NewTextBlockObject(slack.MarkdownType, "*User Properties Alert*", false, false)
+		headerBlockSet = append(headerBlockSet, slack.NewSectionBlock(headerBlock, nil, nil))
+
+		// body
+		var attributeBlocks []*slack.TextBlockObject
+		attributeBlocks = append(attributeBlocks, slack.NewTextBlockObject(slack.MarkdownType, fmt.Sprintf("*Matched User Properties*\n%+v", formattedFields), false, false))
+
+		bodyBlockSet = append(bodyBlockSet, slack.NewSectionBlock(sessionBlock, attributeBlocks, nil))
 	case AlertType.ERROR_FEEDBACK:
```
Comment (SpennyNDaJets) [inline backend/model/model.go:2955]: Screenshot 2023-10-09 at 7 55 41 PM

File: backend/model/model.go [hunk 8]
```diff
@@ -2831,62 +2925,76 @@ func (obj *Alert) sendSlackAlert(ctx context.Context, db *gorm.DB, alertID int,
...
-		blockSet = append(blockSet, slack.NewDividerBlock())
-		msg.Blocks = &slack.Blocks{BlockSet: blockSet}
+		headerBlock := slack.NewTextBlockObject(slack.MarkdownType, fmt.Sprintf("*%s Left Feedback*", identifier), false, false)
+		headerBlockSet = append(headerBlockSet, slack.NewSectionBlock(headerBlock, nil, nil))
+
+		// body
+		feedbackBlock := slack.NewTextBlockObject(slack.MarkdownType, input.CommentText, false, false)
+		bodyBlockSet = append(bodyBlockSet, slack.NewSectionBlock(feedbackBlock, nil, nil))
+		bodyBlockSet = append(bodyBlockSet, slack.NewSectionBlock(sessionBlock, nil, nil))
 	case AlertType.RAGE_CLICK:
```
Comment (SpennyNDaJets) [inline backend/model/model.go:2967]: Screenshot 2023-10-09 at 7 56 06 PM

File: backend/model/model.go [hunk 9]
```diff
@@ -2831,62 +2925,76 @@ func (obj *Alert) sendSlackAlert(ctx context.Context, db *gorm.DB, alertID int,
...
-		textBlock = slack.NewTextBlockObject(slack.MarkdownType, fmt.Sprintf("*Rage Clicks Detected:* %d Recent Occurrences\n\n", *input.RageClicksCount), false, false)
-		blockSet = append(blockSet, slack.NewSectionBlock(textBlock, messageBlock, nil))
-		blockSet = append(blockSet, slack.NewDividerBlock())
-		msg.Blocks = &slack.Blocks{BlockSet: blockSet}
+		headerBlock := slack.NewTextBlockObject(slack.MarkdownType, fmt.Sprintf("*Rage Clicks Detected:* %d Recent Occurrences\n\n", *input.RageClicksCount), false, false)
+		headerBlockSet = append(headerBlockSet, slack.NewSectionBlock(headerBlock, nil, nil))
+
+		// body
+		bodyBlockSet = append(bodyBlockSet, slack.NewSectionBlock(sessionBlock, nil, nil))
 	case AlertType.NEW_SESSION:
```
Comment (SpennyNDaJets) [inline backend/model/model.go:2977]: Screenshot 2023-10-09 at 7 56 35 PM

**Review by SpennyNDaJets — COMMENTED** [review]
Examples

File: backend/model/model.go [hunk 10]
```diff
@@ -2739,32 +2765,69 @@ func (obj *Alert) sendSlackAlert(ctx context.Context, db *gorm.DB, alertID int,
...
+		sessionBlock = slack.NewTextBlockObject(slack.MarkdownType, fmt.Sprintf("*Session* <%s|%s>", sessionLink, sessionText), false, false)
+	}
+
 	switch *obj.Type {
 	case AlertType.ERROR:
 		shortEvent := input.Group.Event
-		if len(input.Group.Event) > 50 {
-			shortEvent = input.Group.Event[:50] + "..."
+		if len(input.Group.Event) > 100 {
+			shortEvent = input.Group.Event[:100] + "..."
```
Comment (Vadman97) [inline backend/model/model.go:2807]: could make this even longer? 100 (1.5 lines as in your example) still seems pretty short

File: backend/model/model.go (hunk 4 above)
Comment (Vadman97) [reply backend/model/model.go:2907]: yes pls! new user alerts altogether?

File: backend/model/model.go (hunk 6 above)
Comment (Vadman97) [reply backend/model/model.go:2939]: should this be `identified_email: true` instead of the json array syntax?

File: backend/model/model.go [hunk 11]
```diff
@@ -2831,62 +2925,76 @@ func (obj *Alert) sendSlackAlert(ctx context.Context, db *gorm.DB, alertID int,
...
-		previewText = "Highlight: Error Feedback Alert"
+		// header
 		if identifier == "" {
 			identifier = "User"
 		}
-		textBlock = slack.NewTextBlockObject(slack.MarkdownType, fmt.Sprintf("*%s Left Feedback*\n\n%s", identifier, input.CommentText), false, false)
-		blockSet = append(blockSet, slack.NewSectionBlock(textBlock, messageBlock, nil))
-		blockSet = append(blockSet, slack.NewDividerBlock())
-		msg.Blocks = &slack.Blocks{BlockSet: blockSet}
+		headerBlock := slack.NewTextBlockObject(slack.MarkdownType, fmt.Sprintf("*%s Left Feedback*", identifier), false, false)
```
Comment (Vadman97) [inline backend/model/model.go:2960]: the fact that this is `feedback` has been confusing for folks and an artifact of an old feature that is deprecated
could we rename this?
```
		headerBlock := slack.NewTextBlockObject(slack.MarkdownType, fmt.Sprintf("*%s Left a Crash Report*", identifier), false, false)
```

**Review by Vadman97 — APPROVED** [review]
implementation is looking great! leaving the final design review to @julian-highlight

File: backend/model/model.go (hunk 4 above)
Comment (SpennyNDaJets) [reply backend/model/model.go:2907]: I have to dig a little deeper, but saw the last new user alert we received in #new-user-alerts was in early September

https://github.com/highlight/highlight/issues/6835

File: backend/model/model.go (hunk 6 above)
Comment (SpennyNDaJets) [reply backend/model/model.go:2939]: Screenshot 2023-10-10 at 9 56 08 AM

File: backend/model/model.go (hunk 10 above)
Comment (SpennyNDaJets) [reply backend/model/model.go:2807]: Screenshot 2023-10-10 at 9 55 48 AM

File: backend/model/model.go (hunk 11 above)
Comment (SpennyNDaJets) [reply backend/model/model.go:2960]: Screenshot 2023-10-10 at 9 58 01 AM

Updated the preview text as well

File: backend/model/model.go (hunk 3 above)
Comment (SpennyNDaJets) [reply backend/model/model.go:2804]: Updated 
Screenshot 2023-10-10 at 10 17 10 AM

**Pull request merged**